### ISSUE TITLE
This response does not support redirection

### ISSUE THREAD

**Issue by Jobinjose01** [thephpleague/omnipay-securepay#5]
When I try to make a purchase it throws an error. like "This response does not support redirection".

what I tried is,

$formData = array(
                    'firstName' => 'Jobin',
                    'lastName' => 'Jose',
                    'number' => '4444333322221111',
                    'expiryMonth'=>'06',
                    'expiryYear'=>'2016',
                    'cvv'=>'123'
                );

```
        $response = $gateway->purchase([
            'cancelUrl' =>'custom_url'),
            'returnUrl' => 'custom_url'),
            'amount' => '1.00',
            'card' => $formData,
            'transactionId' => '123',
            'currency' => 'AUD',
        ])->send();
        if ($response->isRedirect()) {
            $response->redirect();
        } else {
            return 'false';
        }
```

is there any issue with my code ??

**Comment by timeverts** [comment]
I get this issue as well.

The following fix needs to be applied to `src/Message/DirectPostAuthorizeResponse.php` to resolve this:
- Add `use Omnipay\Common\Message\RedirectResponseInterface;` beneath existing `use` statements.
- Change `class DirectPostAuthorizeResponse extends AbstractResponse` to `class DirectPostAuthorizeResponse extends AbstractResponse implements RedirectResponseInterface`

A [pull request](https://github.com/thephpleague/omnipay-securepay/pull/1) was created with this exact fix, but it has never been integrated.

@amacneil, is there any chance you can implement this fix, as it's quite a significant bug?

**Comment by Jobinjose01** [comment]
@timeverts 

I figure out the issue , for me its redirect url is not public , I tested it from my local , so once they trying to post the response the given return url not works.

So try it with public return url. it should work.

Hope it works.

**Comment by amacneil** [comment]
@timeverts I'm no longer the maintainer of this package, but @greydnls or @delatbabel should be able to review and merge this

**Comment by delatbabel** [comment]
This should be a reasonably easy fix. The code to fix this was submitted as a PR and there is another related PR but neither PR can be merged due to the Travis build failing. If someone wants to either resubmit the PRs or amend them so that the builds pass, I can merge them.

I don't have a SecurePay test or production account so it's not possible for me to verify whether the code works, but at least the Travis build needs to pass before I can merge the submitted PRs.

**Comment by delatbabel** [comment]
Merged #10
